qemu/tests/qemu_guest_agent_snapshot.py

```diff
@@ -41,18 +42,39 @@ def _action_before_fsfreeze(self, *args):
             dict(verbose=True, timeout=copy_timeout))
         self.bg.start()
 
+    def check_snapshot(self):
+        """
+        Check whether the snapshot is created successfully.
+        """
+        snapshot_info = str(self.vm.monitor.info("block"))
+        snapshot_node_name = self.params.get("snapshot_node_name")
+        if self.params.get("snapshot_file") not in snapshot_info:
+            raise self.test.fail("Snapshot doesn't exist:%s" % snapshot_info)
+        if snapshot_node_name:
+            match_string = "u?'node-name': u?'%s'" % snapshot_node_name
+            if not re.search(match_string, snapshot_info):
+                self.test.fail("Can not find node name %s of"
+                               " snapshot in block info %s"
+                               % (snapshot_node_name,snapshot_info))
+
+    def cleanup(self, test, params, env):
+        snapshot_create = BlockDevSnapshotTest(self.test, self.params,
```

Comment:
1. This method is not used.
2. You can initialize `BlockDevSnapshotTest` at `def __init__`, so this class can be used anywhere without re-define it here.

Reply:
1. since cleanup will be called in BaseVirtTest.execute, no need to call it in this case
2. update an init method about self.snapshot_create = BlockDevSnapshotTest(self.test, self.params,
                                                    self.env)